data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java

```diff
@@ -0,0 +1,203 @@
+package gov.cdc.etldatapipeline.investigation.util;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import gov.cdc.etldatapipeline.investigation.repository.model.dto.*;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Component;
+
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.util.*;
+
+@Component
+@RequiredArgsConstructor
+public class ProcessInvestigationDataUtil {
+    private static final Logger logger = LoggerFactory.getLogger(ProcessInvestigationDataUtil.class);
+
+    @Value("${spring.kafka.stream.output.investigation.topic-name-confirmation}")
+    public String investigationConfirmationOutputTopicName = "cdc.nbs_odse.dbo.Investigation.Confirmation";
+
+    @Value("${spring.kafka.stream.output.investigation.topic-name-notification}")
+    public String investigationNotificationOutputTopicName = "cdc.nbs_odse.dbo.Investigation.Notification";
+
+    @Value("${spring.kafka.stream.output.investigation.topic-name-observation}")
+    public String investigationObservationOutputTopicName = "cdc.nbs_odse.dbo.Investigation.output.Observation";
+
+    private final KafkaTemplate<String, String> kafkaTemplate;
+
+    public InvestigationTransformed transformInvestigationData(Investigation investigation) {
+
+        InvestigationTransformed investigationTransformed = new InvestigationTransformed();
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        transformPersonParticipations(investigation.getPersonParticipations(), investigationTransformed, objectMapper);
+        transformOrganizationParticipations(investigation.getOrganizationParticipations(), investigationTransformed, objectMapper);
+        transformActIds(investigation.getActIds(), investigationTransformed, objectMapper);
+        transformObservationNotificationIds(investigation.getObservationNotificationIds(), investigationTransformed, objectMapper);
+        transformInvestigationConfirmationMethod(investigation.getInvestigationConfirmationMethod(), investigationTransformed, objectMapper);
+
+        return investigationTransformed;
+    }
+
+
+    private void transformPersonParticipations(String personParticipations, InvestigationTransformed investigationTransformed, ObjectMapper objectMapper) {
+        try {
+            JsonNode personParticipationsJsonArray = personParticipations != null ? objectMapper.readTree(personParticipations) : null;
```

Comment:
Consider creating a java model for the post processing and doing the transformations based on that.

Reply:
We already have `Investigation.java` which contains the data from the stored proc and doing post processing based on that.